py/builtinimport.c

```diff
@@ -145,11 +145,16 @@ STATIC void do_load_from_lexer(mp_obj_t module_obj, mp_lexer_t *lex) {
 #endif
 
 #if MICROPY_PERSISTENT_CODE_LOAD || MICROPY_MODULE_FROZEN_MPY
-STATIC void do_execute_raw_code(mp_obj_t module_obj, mp_raw_code_t *raw_code) {
+STATIC void do_execute_raw_code(mp_obj_t module_obj, mp_raw_code_t *raw_code
     #if MICROPY_PY___FILE__
```

Comment:
These conditionals shouldn't be necessary.  This function is static so the compiler is (usually) smart enough to know not to pass this argument if it's not needed.  Indeed compiling this without this guard gives the same code size for all ports.

Reply:
> (usually)
isn't that a bit risky then ?
on new platform compiler are not always smart at the beginning.. 
also did you test all compilers on win32 platform, it seems there's a zoo in there.

Reply:
It's not a problem if the compiler cannot optimise it, it'll just be a little bit larger in code size.  This "trick" is already used in the code base, eg `py/objtype.c:check_for_special_accessors`

Reply:
@pmp-p One extra parameter is not going to increase code size by any significant amount, especially since it's never used.